### ISSUE TITLE
Parameter "url" support also with http?://

### ISSUE THREAD

**Issue by kleisauke** [andrieslouw/imagesweserv#19]
# Parameter "url" support also with http?:// [16592074] #

**Submitted by Binyamin on 10-10-2016 0:00:00**  
**1 votes on UserVoice prior to migration**  

I propose to allow parameter "url" support also http:// and https://, there are many reasons for it.
1. more consistent API with similar services
2. http and https may respond different results, for example https://a.b/kitties.jpg may return cats image and http://a.b/kitties.jpg error/fallback image or error 404, or else
3. when defined https://... would reduce the server-side loop - wouldn't try check http:// before trying https:// (for example domain might support only https)
I am fine with to continue to support "url" value without protocol (maybe as a backwards compatibility).

## Response ##
**by Andries Louw Wolthuizen on 11-10-2016 0:00:00**

As stated on the frontpage, SSL and HTTP is already supported since january 2013: /archive/suggestion-2693328-add-support-to-fetch-images-over-https
The API is inconsistent to prevent UBB-parsers to fail on the double http://-part. For that reason we use ssl:.
More detailed answer on your questions:
1. See answer above.
2. Yes, sadly, in many cases we tested, HTTPS is wildly different from HTTP. Looking for both increases the parsing time needed to fetch the requested image. Also, for us it doesn’t make sense to always fetch over HTTPS, it’ll only increase lag, and won’t add any security (we could be considered to be a middle-man, and we would only protect against additional middle-mans between our datacenter, and that of the origin. If there is such an attack, SSL wouldn’t help us.).
3. There is no server-side loop, see answer to 1. We only check http:// for ?url=, and only check https:// for ?url=ssl:…-requests.
And indeed, our system is already backwards-compatible for more than 8 years, so changing such a thing would be a real hassle.
To summarize: The only new thing in your suggestion is to replace ?url= for ?url=http:// and ?url=ssl: for ?url=https://. This is something we consider as being bad-practice, therefore it is declined.

**[Original UserVoice Submission](https://imagesweserv.uservoice.com/forums/144259-images-weserv-nl-general/suggestions/16592074)**

**Comment by andrieslouw** [comment]
See https://github.com/andrieslouw/imagesweserv/issues/33
